modules/backend/widgets/Lists.php

```diff
@@ -525,18 +525,24 @@ public function prepareQuery()
          * Apply sorting
          */
         if (($sortColumn = $this->getSortColumn()) && !$this->showTree && in_array($sortColumn, array_keys($this->getVisibleColumns()))) {
-            if (($column = array_get($this->allColumns, $sortColumn)) && $column->valueFrom) {
-                $sortColumn = $this->isColumnPivot($column)
-                    ? 'pivot_' . $column->valueFrom
-                    : $column->valueFrom;
-            }
+            if ($this->sortColumnHasCustomSorting()) {
+                $customSort = $this->getColumns()[$this->getSortColumn()]->sortable;
 
-            // Set the sorting column to $relation_count if useRelationCount enabled
-            if (isset($column->relation) && @$column->config['useRelationCount']) {
-                $sortColumn = $column->relation . '_count';
-            }
+                $this->buildQueryCustomSort($query, $customSort);
+            } else {
+                if (($column = array_get($this->allColumns, $sortColumn)) && $column->valueFrom) {
```

Comment:
@bradietilley One thing I try to avoid is those variable definitions inside condition calls. I find Xdebug generally has trouble getting those variables until *after* the condition is processed, when sometimes you want to know what it is *before* the condition runs.
```suggestion
                $column = array_get($this->getColumns(), $sortColumn);
                if ($column && $column->valueFrom) {
```

Reply:
Also, if I'm not mistaken, the `$sortColumn` on this line is no longer defined with your changes in place.